akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -1194,10 +1215,14 @@ class DDataShardCoordinator(
   // which was scheduled by previous watchStateActors
   def waitingForStateInitialized: Receive = {
     case StateInitialized =>
-      unstashGetShardHomeRequests()
-      unstashAll()
-      stateInitialized()
-      activate()
+      log.debug("StateInitialized") // FIXME remove
+      update(ShardCoordinatorInitialized) { _ =>
```

Comment:
A comment explaining how this write ensures consistency would be good if we decide to keep it

Reply:
I'll think another round if we need this. Probably not.

Reply:
reverted this since I don't think there is a strong reason to change it with the new additional reads